Trivy errors on some builds

Seeing the following errors with the new Trivy scans, causing some builds to fail

```
Running Trivy with options: trivy image ghcr.io/stacklok/dockyard/npx/sentry-mcp-server:0.23.1
2025-11-26T03:42:06Z	INFO	[secret] Secret scanning is enabled
2025-11-26T03:42:06Z	INFO	[secret] If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2025-11-26T03:42:06Z	INFO	[secret] Please see also https://trivy.dev/v0.65/docs/scanner/secret#recommendation for faster secret detection
📣 Notices:
  - Version 0.67.2 of Trivy is now available, current version is 0.65.0
To suppress version checks, run Trivy scans with the --skip-version-check flag
2025-11-26T03:42:06Z	FATAL	Fatal error	run error: image scan error: scan error: unable to initialize a scan service: unable to initialize an image scan service: unable to find the specified image "ghcr.io/stacklok/dockyard/npx/sentry-mcp-server:0.23.1" in ["docker" "containerd" "podman" "remote"]: 4 errors occurred:
	* docker error: unable to inspect the image (ghcr.io/stacklok/dockyard/npx/sentry-mcp-server:0.23.1): Error response from daemon: No such image: ghcr.io/stacklok/dockyard/npx/sentry-mcp-server:0.23.1
	* containerd error: failed to list images from containerd client: connection error: desc = "transport: Error while dialing: dial unix /run/containerd/containerd.sock: connect: permission denied"
	* podman error: unable to initialize Podman client: no podman socket found: stat /run/user/1001/podman/podman.sock: no such file or directory
	* remote error: GET https://ghcr.io/v2/stacklok/dockyard/npx/sentry-mcp-server/manifests/0.23.1: MANIFEST_UNKNOWN: manifest unknown
Error: Process completed with exit code 1.
```

For example, two builds failing on #213 (sentry and graphlit). On #212 it's failing on phoenix-mcp.

I suspect it's trying to pull an image that hasn't been published yet, instead of checking the local image. Sentry's last build on `main` failed; the phoenix PR is bumping that one so the new image won't be pushed until it's merged.

--- a/.github/workflows/build-containers.yml
+++ b/.github/workflows/build-containers.yml
@@ -377,6 +377,17 @@ jobs:
           sbom: true
           provenance: true
 
+      - name: Build single-platform image for Trivy scan
+        id: build-for-scan
+        uses: docker/build-push-action@263435318d21b8e681c14492fe198d362a7d2c83 # v6
+        with:
+          context: ${{ steps.dockerfile.outputs.dockerfile_dir }}
+          file: ${{ steps.dockerfile.outputs.dockerfile_path }}
+          platforms: linux/amd64
+          load: true
+          tags: local-scan:${{ steps.meta.outputs.server_name }}-${{ steps.meta.outputs.version }}
+          cache-from: type=gha
+
       - name: Sign container images with Cosign
         if: github.event_name != 'pull_request'
         env:
@@ -450,7 +461,7 @@ jobs:
       - name: Run Trivy vulnerability scanner
         uses: aquasecurity/trivy-action@b6643a29fecd7f34b3597bc6acb0a98b03d33ff8 # 0.33.1
         with:
-          image-ref: ${{ steps.meta.outputs.image_name }}:${{ steps.meta.outputs.version }}
+          image-ref: local-scan:${{ steps.meta.outputs.server_name }}-${{ steps.meta.outputs.version }}
           format: 'sarif'
           output: 'trivy-results.sarif'
           severity: 'CRITICAL,HIGH,MEDIUM'

